packages/code-studio/src/settings/FormattingSectionContent.tsx

```diff
@@ -201,199 +197,135 @@ export class FormattingSectionContent extends PureComponent<
     event: ChangeEvent<HTMLSelectElement>
   ): void {
     log.debug('handleDefaultDateTimeFormatChange', event.target.value);
-    this.setState(
-      {
-        defaultDateTimeFormat: event.target.value,
-      },
-      () => {
-        this.debouncedCommitChanges();
-      }
-    );
+    const update = {
+      defaultDateTimeFormat: event.target.value,
+    };
+    this.setState(update);
+    this.commitChanges(update);
   }
 
   handleDefaultDecimalFormatChange(event: ChangeEvent<HTMLInputElement>): void {
     log.debug('handleDefaultDecimalFormatChange', event.target.value);
-    this.setState(
-      {
-        defaultDecimalFormatOptions: {
-          defaultFormatString: event.target.value,
-        },
+    const update = {
+      defaultDecimalFormatOptions: {
+        defaultFormatString: event.target.value,
       },
-      () => {
-        this.debouncedCommitChanges();
-      }
-    );
+    };
+    this.setState(update);
+    if (
+      isValidFormat(
+        dh,
+        TableUtils.dataType.DECIMAL,
+        DecimalColumnFormatter.makeCustomFormat(event.target.value)
+      )
+    ) {
+      this.commitChanges(update);
+    }
   }
 
   handleDefaultIntegerFormatChange(event: ChangeEvent<HTMLInputElement>): void {
     log.debug('handleDefaultIntegerFormatChange', event.target.value);
-    this.setState(
-      {
-        defaultIntegerFormatOptions: {
-          defaultFormatString: event.target.value,
-        },
+    const update = {
+      defaultIntegerFormatOptions: {
+        defaultFormatString: event.target.value,
       },
-      () => {
-        this.debouncedCommitChanges();
-      }
-    );
+    };
+    this.setState(update);
+    if (
+      isValidFormat(
+        dh,
+        TableUtils.dataType.INT,
+        IntegerColumnFormatter.makeCustomFormat(event.target.value)
+      )
+    ) {
+      this.commitChanges(update);
```

Comment:
We should still debounce these changes, as they could have implications for the entire UI (e.g. if you have a table open and you're updating the formatting, it'll update every cell of every table open with every keystroke).
See my other comment in this file about debouncing for a suggestion.